Assets/MRTK/SDK/Features/UX/Scripts/PulseShader/HandPulseLogic.cs

```diff
@@ -0,0 +1,244 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Linq.Expressions;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI.SurfacePulse
+{
+    /// <summary>
+    /// Script for triggering the pulse shader effect on hand mesh.
+    /// </summary>
+    [AddComponentMenu("Scripts/MRTK/SDK/HandPulseLogic")]
+    [RequireComponent(typeof(SurfacePulse))]
```

Comment:
This also feels a little weird. Since `SurfacePulse` has that explicit logic about the spatial mapping, it feels possible to accidentally get into a state where i want to turn on pulse in the hand mesh but accidentally turn on pulse for spatial mapping as well.

Reply:
@keveleigh It does seem like there is a potential for confusion with the current structure.

What if instead we can create and move the core pulse shader logic to a base class and then we could extend this class with a HandMeshPulseShaderHandler and SpatialMeshPulseShaderHandler components?  
This way we would have separate components and a distinct separation of the logic required to handle the hand mesh vs spatial mesh.

Reply:
Seems reasonable to me!

Reply:
@cre8ivepark how does the structure rework sound to you?

Reply:
It sounds great!